Sources/DVR/SessionDataTask.swift

```diff
@@ -42,42 +42,50 @@ final class SessionDataTask: URLSessionDataTask {
     override func resume() {
 
         // apply request transformations, which could impact matching the interaction
-        var filteredRequest = request
-        session.filters.forEach { filter in
-            filteredRequest = filter.filter(request: filteredRequest)
+        var filteredRequest: URLRequest? = request
+        for filter in session.filters {
+            if let req = filteredRequest {
+                filteredRequest = filter.filter(request: req)
+            }
+            else {
+                break
+            }
         }
```

Comment:
This logic looks very similar for response. It might be worthwhile to extract these to helper methods on `Session`. This would make the code more readable and maintainable. It also eliminate the `break`:
```swift
func filter(request: URLRequest) -> URLRequest? {
  var filteredRequest: URLRequest? = request
  for filter in self.filters {
    guard let req = filteredRequest else { return nil }
    filteredRequest = filter.filter(request: req)
  }
  return filteredRequest
}
```

Reply:
For response it seems simpler to leave it as is, otherwise I'd have to add an extra return value to pass along the bool for `persistInteraction`. Should I still extract it to Session?